lzss/compress.go

```diff
@@ -95,47 +110,41 @@ func InitBackRefTypes(dictLen int, level Level) (short, long, dict BackrefType)
 		}
 	}
 	short = newBackRefType(SymbolShort, wordAlign(14), 8, false)
-	long = newBackRefType(SymbolLong, wordAlign(19), 8, false)
+	long = newBackRefType(SymbolLong, wordAlign(longBrAddressNbBits), 8, false)
 	dict = newBackRefType(SymbolDict, wordAlign(bits.Len(uint(dictLen))), 8, true)
 	return
 }
 
-// Compress compresses the given data; if hint is provided, the compressor will try to use it
-// hint should be a subset of the data compressed by the same compressor
-// For example, calling Compress([]byte{1, 2, 3, 4, 5}, compressed([]byte{1, 2, 3})) will
-// result in much faster compression than calling Compress([]byte{1, 2, 3, 4, 5})
-func (compressor *Compressor) Compress(d []byte, hints ...[]byte) (c []byte, err error) {
+// The compressor cannot recover from a Write error. It must be Reset before writing again
+func (compressor *Compressor) Write(d []byte) (n int, err error) {
 	// check input size
 	if len(d) > MaxInputSize {
-		return nil, fmt.Errorf("input size must be <= %d", MaxInputSize)
+		return 0, fmt.Errorf("input size must be <= %d", MaxInputSize)
 	}
 
-	// reset output buffer
-	compressor.buf.Reset()
-
-	// write header
-	header := Header{Version: Version, Level: compressor.level}
-	if _, err = header.WriteTo(&compressor.buf); err != nil {
+	// reconstruct bit writer cache
+	compressor.lastOutLen = compressor.outBuf.Len()
+	lastByte := compressor.outBuf.Bytes()[compressor.outBuf.Len()-1]
+	compressor.outBuf.Truncate(compressor.outBuf.Len() - 1)
+	lastByte >>= compressor.nbSkippedBits
+	if err = compressor.bw.WriteBits(uint64(lastByte), 8-compressor.nbSkippedBits); err != nil {
 		return
 	}
 
+	compressor.lastNbSkippedBits = compressor.nbSkippedBits
+	compressor.appendInput(d)
+
 	// write uncompressed data if compression is disabled
 	if compressor.level == NoCompression {
-		compressor.buf.Write(d)
-		return compressor.buf.Bytes(), nil
+		compressor.outBuf.Write(d)
+		return len(d), nil
 	}
 
+	n, d = len(d), compressor.inBuf.Bytes()
+
 	// initialize bit writer & backref types
-	compressor.bw = bitio.NewWriter(&compressor.buf)
 	shortBackRefType, longBackRefType, dictBackRefType := InitBackRefTypes(len(compressor.dictData), compressor.level)
 
-	startI := 0
-	if len(hints) == 1 {
-		// try to compress from the hint to save time (no need to look for optimal backrefs
-		// if we already have a good enough hint)
-		startI = compressor.compressFromHint(header, d, hints[0])
-	}
-
 	// build the index
 	compressor.inputIndex = suffixarray.New(d, compressor.inputSa[:len(d)])
```

Comment:
seems the inputIndex is built only on the last (len(d)) chunk? So not looking for backrefs in the previously appended chunks?